### DISCUSSION TITLE
Debian 12 - IP is banned (f2b log) but not blocked

### DISCUSSION THREAD

**Discussion by walhallaRV** [fail2ban/fail2ban#3583] [General]
Hi,
F2B seems to work properly. I can see that it finds bad logins and bans it (example FTP here):

fail2ban.log ->
...
2023-09-26 23:04:11,854 fail2ban.filter         [1688703]: INFO    [proftpd] Found 47.108.92. ** - 2023-09-26 23:04:11
2023-09-26 23:04:16,079 fail2ban.filter         [1688703]: INFO    [proftpd] Found 47.108.92. ** - 2023-09-26 23:04:16
2023-09-26 23:04:20,011 fail2ban.filter         [1688703]: INFO    [proftpd] Found 47.108.92. ** - 2023-09-26 23:04:20
2023-09-26 23:04:20,805 fail2ban.actions        [1688703]: WARNING [proftpd] 47.108.92. ** already banned

I was wondering why it finds this IP again and again if it is banned? So I looked at firewalld:

Direct	IPv4	47.108.92. **	Input	Reject	ipv4 filter INPUT_direct 0 -p tcp -m multiport --dports ftp,ftp-data,ftps,ftps-data -m set --match-set f2b-proftpd src -j REJECT --reject-with icmp-port-unreachable

and (ipset list) ->

Name: f2b-proftpd
Type: hash:ip
Revision: 5
Header: family inet hashsize 1024 maxelem 65536 timeout 0 bucketsize 12 initval 0x7c780ec4
Size in memory: 264
References: 1
Number of entries: 1
Members:
47.108.92. ** timeout 0

So, this IP should be blocked - but i still can see logins every second (proftp.log) ->
...
2023-09-26 23:09:20,381 proftpd[1711663] 0.0.0.0 (47.108.92. **[47.108.92. **]): USER root (Login failed): Incorrect password
2023-09-26 23:09:25,248 proftpd[1711664] 0.0.0.0 (47.108.92. **[47.108.92. **]): USER root (Login failed): Incorrect password
2023-09-26 23:09:29,123 proftpd[1711665] 0.0.0.0 (47.108.92. **[47.108.92. **]): USER root (Login failed): Incorrect password
...

banaction is ->

[DEFAULT]
banaction = firewallcmd-ipset

So, this IP was detected, banned, blocked -> but still can try to login (brute & force)? 

What is going wrong? Fresh installed Debian 12 (bookworm) all packages up to date.

Thanks for any hint / tipp / or even better: solution ;)

Walhalla

**Comment by sebres** [comment]
See the answer for a 2nd question in [wiki : FAQ](https://github.com/fail2ban/fail2ban/wiki/How-fail2ban-works)...

I have nothing to add here right now, excepting possibly:  either wrong/missing port in jail (other as `ftp,ftp-data,ftps,ftps-data`) or some white-listing rules in FW before INPUT_direct allowing packets from that IP.

**Comment by walhallaRV** [comment]
Hi and thanks for your response!
I already read the WIKI/FAQ. But nothing described there is the problem. Yes, FTP is using normal / standard ports and the ports are open to connect on the FW. But FTP was just an example. I have the same problem with postfix/sasl, IMAP, etc. IPs detected, banned, "blocked" but still able to connect.

Maybe there is an error generating the rule (action)? I tried and added / blocked manually ->

firewall-cmd --add-rich-rule='rule family=ipv4 source address=47.108.92. ** reject' --permanent

and this boring idiot was blocked immediately. The log stopped, no more connects from this IP?!

Afterwords i tested an changed from banaction firewall-cmd to nftables. This works. So it seams that maybe some error in the regarding action?

**Reply by sebres** [reply]
> Afterwords i tested an changed from banaction firewall-cmd to nftables. This works. 

Then use it instead of firewall-cmd! Never understand why such intermediate layers are needed.

> So it seams that maybe some error in the regarding action?

No idea, I never used firewalld and am not quite familiar with it... Maybe it is the same issue like https://github.com/fail2ban/fail2ban/issues/1609#issuecomment-303085942: firewalld seems to have an issue with iptables backend, so if I understand correctly a switch of firewalld backend to nftables may help, quote from https://github.com/firewalld/firewalld/issues/44#issuecomment-408211978:
*This was addressed with the new nftables backend (v0.6.0). Direct rules have higher precedence and occur before any other firewalld rules. For more info please read the blog post*

**Comment by walhallaRV** [comment]
> Then use it instead of firewall-cmd! Never understand why such intermediate layers are needed.

YES. I switched to nftables. I used because many sayed that it is more performant if you have big tables of blocked IPs. 


> No idea, I never used firewalld and am not quite familiar with it...

Neighter I. Too old to switch and really prefer / love iptables ;))))

Thanks for your help. Simply will stay without this firewalld! Never change a working system.

Cheers Walhalla

**Reply by sebres** [reply]
> I used because many sayed that it is more performant if you have big tables of blocked IPs.

Did they mean ipset probably? Because nftables has its own sets too that are comparable to iptables+ipset combi (what basically firewalld uses internally by its iptables backend)... 
Anyway I cannot follow why firewalld should be faster, and how it can be if it is just a wrapper over native net-filter.

With other words: filtering via iptables pure is slower than iptables+ipset, and nftables (with sets) shows comparable performance. Whereas firewalld is just a new layer over both native filters.
